dicom-archive/updateMRI_Upload.pl

```diff
@@ -109,81 +123,118 @@
 ################################################################
 unless (-e $source_location) {
     print "\nERROR: Could not find sourcelocation $source_location \nPlease,
-           make sure the sourcelocation is correct. Upload will 
+           make sure the sourcelocation is correct. Upload will
            exit now.\n\n\n";
     exit 5;
 }
 ################################################################
 #####establish database connection if database option is set####
 ################################################################
-my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db); 
+my $db = NeuroDB::Database->new(
+    databaseName => $Settings::db[0],
+    userName     => $Settings::db[1],
+    password     => $Settings::db[2],
+    hostName     => $Settings::db[3]
+);
+
 print "Connecting to database.\n" if $verbose;
 
+try {
+    $db->connect();
+} catch(NeuroDB::DatabaseException $e) {
+    die sprintf(
+        "User %s failed to connect to %s on %s: %s (error code %d)\n",
+        $Settings::db[1],
+        $Settings::db[0],
+        $Settings::db[3],
+        $e->errorMessage,
+        $e->errorCode
+    );
+}
+
 ################################################################
 #####check to see if the tarchiveid is already set or not#######
 ######if it's already in the mri_upload table then it will######
 ######generate an error#########################################
 ################################################################
 
 # fetch tarchiveLibraryDir from ConfigSettings in the database
-my $tarchiveLibraryDir = &NeuroDB::DBI::getConfigSetting(
-                            \$dbh,'tarchiveLibraryDir'
-                            );
-# determine tarchive path stored in the database (without tarchiveLibraryDir)
-my $tarchive_path = $tarchive;
-$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;  
 
-my $where         = " WHERE t.ArchiveLocation =?";
+my $configOB = NeuroDB::objectBroker::ConfigOB->new(db => $db);
 
-if ($globArchiveLocation) {
-    $where = " WHERE t.ArchiveLocation LIKE ?";
-    $tarchive_path = basename($tarchive);
+my $tarchiveLibraryDir;
+try {
+    $tarchiveLibraryDir = $configOB->getTarchiveLibraryDir();
+} catch(NeuroDB::objectBroker::ObjectBrokerException $e) {
+    die sprintf(
+        "Failed to get tarchive library dir from config: %s\n",
+        $e->errorMessage
+    );
+}
+# determine tarchive path stored in the database (without tarchiveLibraryDir)
+my $tarchive_path = $tarchive;
+$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;
+
+# Check if there is already an mri upload record for the tarchive
+my $resultRef;
+my $mriUploadOB = NeuroDB::objectBroker::MriUploadOB->new(db => $db);
+try{
+    $resultRef = $mriUploadOB->getWithTarchive(
+        GET_COUNT, $tarchive_path, $globArchiveLocation
```

Comment:
Get the following error here:
"Failed to retrieve mri upload records: called with 1 bind variables when 0 are needed"

The query looks OK I think: 
`SELECT COUNT(*) FROM mri_upload JOIN tarchive USING(TarchiveID) WHERE ArchiveLocation LIKE "%?%" `
@args = 'DCM_2012-11-06_MTL0002_999145_NAPBL00_20121106_141430836.tar'

This should return Count(*) = 0 but instead I get the error above...

Reply:
Yikes! Good catch! Will correct.